pyccel/codegen/printing/ccode.py

```diff
@@ -443,7 +455,7 @@ def _handle_is_operator(self, Op, expr):
         a = expr.args[0]
         b = expr.args[1]
 
-        if (a.dtype is NativeBool() and b.dtype is NativeBool()):
+        if (Nil() not in expr.args and a.dtype is NativeBool() and b.dtype is NativeBool()):
```

Comment:
Maybe it would be simpler to reverse the order of these `If` statements rather than testing for `Nil()` twice